File: codeWars/5kyu/TheHashtagGenerator/python/theGeneratorHashtag.py
"""The Generator Hashtag"""

def generator_hashtag(s):
    if len(s) == 0:
        return False
    s = s.strip()
    s = s.title()
    s = s.split()
    s = ''.join(s)
    s = "#" + s
    if len(s) > 140 or len(s) == 0:
        return False
    return s

# Testing
s0 = " Hello there thanks for trying my Kata"
s1 = "    Hello     World   "
s2 = ""
print(generator_hashtag(s0))

'''Complexity Analysis:
Time Complexity: O(N) where N is the length of the array
Space Complexity: O(1)
'''
t = "ehe leke eltjl"
print(t.title())

File: codeWars/5kyu/TheHashtagGenerator/js/theGeneratorHashtag.js
"use strict";

var generatorHashtag = function(str) {
    str = str.trim();
    if (str.length === 0) {
        return false;
    };
    str = str.split(" ");
    str = str.map(word => {
        return word.charAt(0).toUpperCase() + word.slice(1);
    });
    str = str.join("");
    str = "#" + str;
    if (str.length > 140 || str.length === 0) {
        return false;
    };
    return str;
};

// Testing
let s0 = " Hello there thanks for trying my Kata      ";
let s1 = "    Hello     World   ";
let s2 = "     ";
console.log(generatorHashtag(s2));

// Complexity Analysis:
// Time Complexity: O(N) where N is the length of the array
// Space Complexity: O(1)
